rest_docs/rest-catalog-open-api-v0.1.yaml

```diff
@@ -0,0 +1,657 @@
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/RenameTableRequest'
+        required: true
+      responses:
+        '200':
+          description: OK
+        '401':
+          description: Unauthorized
+        '404':
+          description:
+            Not Found
+            - NoSuchTableException, Table to rename does not exist
+            - NoSuchNamespaceException, The target namespace of the new table identifier does not exist
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/responses/NoSuchTableResponse'
+                  - $ref: '#/components/responses/NoSuchNamespaceResponse'
+              examples:
+                table_to_rename_does_not_exist:
+                  value: {
+                    "error": {
+                      "message": "Table to rename does not exist",
+                      "type": "NoSuchTableException",
+                      "code": 404
+                    } }
+                namespace_to_rename_to_does_not_exist:
+                  value: {
+                    "error": {
+                      "message": "Namespace to rename to does not exist",
+                      "type": "NoSuchNameSpaceException",
+                      "code": 404
+                    } }
+        '406':
+          description: Not Acceptable (UnsupportedOperationException)
+        '409':
+          description: Conflict (AlreadyExistsException - The new target table identifier already exists)
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/TableAlreadyExistsError'
+              example: {
+                "error": {
+                  "message": "Table already exists",
+                  "type": "AlreadyExistsException",
+                  "code": 409
+                } }
+
+components:
+
+  ##############################
+  # Application Schema Objects #
+  ##############################
+  schemas:
+    ResponseDataObject:
+      type: object
+      description: JSON data payload returned in a successful response body
+      properties:
+        data:
+          type: object
+          description: Wrapper for the response of a successful request
+      example: { "data": { "identifiers": [ "office.employees", "office.dogs", "office.cats" ] } }
+    ResponseErrorObject:
+      type: object
+      description: JSON error payload returned in a response with further details on the error
+      required:
+        - message
+        - type
+        - code
+      properties:
+        message:
+          type: string
+          description: Human-readable error message
+        type:
+          type: string
+          description: Internal type of the error, such as an exception class
+          example: NoSuchNamespaceException
+        code:
+          type: integer
+          minimum: 100
```

Comment:
If it were the case that `error` and `data` were both allowed (for example to represent partial failure / partial success), then maybe this would go to 200.

But yeah, most likely we don't expect this to go below 400.

Reply:
Updated to 400.